superset/views/core.py

```diff
@@ -3035,17 +3037,21 @@ class CssTemplateAsyncModelView(CssTemplateModelView):
 appbuilder.add_link(
     'Query Search',
     label=_('Query Search'),
-    href='/superset/sqllab#search',
+    href='Superset.sqllab',
     icon='fa-search',
     category_icon='fa-flask',
     category='SQL Lab',
     category_label=__('SQL Lab'),
 )
 
+qs_menu_item = appbuilder.menu.find('Query Search')
+qs_get_url = qs_menu_item.get_url
+qs_menu_item.get_url = lambda: qs_get_url() + '#search'
+
```

Comment:
can we find a way to pass params in url_for method ....instead setting it manually

Reply:
As discussed with @ankursinghal2005 this is how we can handle links with `href`. If we find better resolution in near future, we can adopt the same.